gcsio/src/main/java/com/google/cloud/hadoop/gcsio/GoogleCloudStorageClientImpl.java

```diff
@@ -218,13 +223,48 @@ private static Storage createStorage(
               return ImmutableList.copyOf(list);
             })
         .setCredentials(credentials != null ? credentials : NoCredentials.getInstance())
-        .setBlobWriteSessionConfig(
-            BlobWriteSessionConfigs.getDefault()
-                .withChunkSize(storageOptions.getWriteChannelOptions().getUploadChunkSize()))
+        .setBlobWriteSessionConfig(getSessionConfig(storageOptions.getWriteChannelOptions()))
         .build()
         .getService();
   }
 
+  private static BlobWriteSessionConfig getSessionConfig(AsyncWriteChannelOptions writeOptions)
+      throws IOException {
+    BlobWriteSessionConfig bufferConfig;
+    logger.atFiner().log("Upload strategy in use: %s", writeOptions.getUploadType());
+    switch (writeOptions.getUploadType()) {
```

Comment:
nit: If with return might be more readable.

Reply:
updated.

Reply:
I prefer switch over if-else logic here.